### DISCUSSION TITLE
How to add colorscheme to Wezterm

### DISCUSSION THREAD

**Discussion by ribru17** [wezterm/wezterm#3850] [Q&A] [answered]
Hello, I have a `.toml` file that represents my color scheme and I would like to add it to Wezterm. Researching online I found that there have been several different ways of getting this to happen, such as uploading it to the iterm2 repo or making a separate repo and syncing that one directly. When I see the color schemes on the iterm2 repo they seem to be missing many fields that I want my color scheme to have (e.g. `[colors.tab_bar.new_tab_hover]`) and I don't know if this is intentional and if this extra information should be stored somewhere else. What is the best way to go about this? Thanks for any help.

**Comment by wez** [comment, marked as answer]
In theory, it's as simple as this portion of this commit:

https://github.com/wez/wezterm/commit/dd8d4a04f423cd4c96dac825dcf2b489c301a003#diff-e72cae52cc3664b84e22c24186a4804703ac4adf91d6f09315f0af24d6bd91e7R387

however, sometimes the big aggregating repos like Gogh or iterm2 color schemes make changes that are a bit awkward for the versioning stuff in the sync machinery to handle.

That is the case at the moment, and I have https://github.com/wez/wezterm/issues/3831 open to track doing that and adding your and one other scheme to the list.

**Reply by ribru17** [reply]
I see, thank you so much for the reply!

**Comment by mangkoran** [comment]
Hi @wez. Thank you for your work on Wezterm! I would also like to ask regarding adding color scheme to Wezterm. Is there any convention on how we should put the theme file in theme's upstream/own repo? 

I also looked at https://github.com/wez/wezterm/blob/dbed5017f828e8719ad871a031acd4c09c0c98d0/sync-color-schemes/src/main.rs but I still a bit confused how the script works. It seems it will automatically find `.toml` file and process it but that's as far as I understand. CMIIW.

**Reply by wez** [reply]
The importer will try to parse all toml files in the source repo as a color scheme and include whatever it parses, so there's no firm convention on placement within the repo.

I personally enjoy the simplicity of this one:
https://github.com/hardhackerlabs/theme-wezterm

but you can, of course, use directories to structure things more nicely if you have more than one scheme to add.

**Reply by mangkoran** [reply]
I see. So based on that, from what I understand so far the importer will then try to detect if the content of each toml file is a valid color scheme or not via https://github.com/wez/wezterm/blob/dbed5017f828e8719ad871a031acd4c09c0c98d0/sync-color-schemes/src/main.rs#L325C50-L325C55. Is this the case?

I am still trying to understand the inner workings of `sync-color-schemes` 😅

**Reply by wez** [reply]
correct!
